### ISSUE TITLE
SSM agent not discovered with default steps

### ISSUE THREAD

**Issue by rverma-jm** [bottlerocket-os/bottlerocket#914]
While following the guide here. https://github.com/bottlerocket-os/bottlerocket/blob/develop/QUICKSTART.md

*The below steps are for region `ap-south-1`*.
 I had copied the ami to `ap-south-1` using
```aws ec2 copy-image --source-region us-east-1 --source-image-id ami-07fc2dd9e0d7741ed --name bottle-rocket```

Created a launch-template with userdata as defined in above steps. I can get the new instance being discovered in our existing kubernetes cluster, and in Ready state. 

Also currently all of our instances are without any ssh key associated, and we did same for bottle-rocket.

Everything seems fine, but in SSM no new instances reflected in managed instances.

Userdata.toml = 
```
[settings.kubernetes]
api-server = "https://l4.ap-south-1.eks.amazonaws.com"
cluster-certificate = "cert"
cluster-name = "cluster-name"
```
Basically the result of
```
eksctl get cluster --region us-west-2 --name bottlerocket -o json \
   | jq --raw-output '.[] | "[settings.kubernetes]\napi-server = \"" + .Endpoint + "\"\ncluster-certificate =\"" + .CertificateAuthority.Data + "\"\ncluster-name = \"bottlerocket\""' > userdata.toml
```

As per docs, control container should be enabled by default, this means SSM agent should be working by default.

Am I missing something?

**Comment by jhaynes** [comment]
There is one extra wrinkle if you're deploying Bottlerocket to new regions. Some of Bottlerocket's default settings are "dynamic" in they are only determined after the instance is running. You can see this behavior in [defaults.toml](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/defaults.toml#L67-L69).

As such, if you don't override the default, Bottlerocket tries to pull the control container from the region you're in. Instead, you can set `settings.host-containers.control.source` via user data to specify either our ECR repos in a currently supported region or a custom endpoint that you control where the control container can be found.

**Comment by rverma-jm** [comment]
@jhaynes I did add below config to userdata in addition to  `settings.kubernetes`

```
[metadata.settings.host-containers.control.source]
setting-generator = "schnauzer settings.host-containers.control.source"
template = "328549459982.dkr.ecr.us-east-1.amazonaws.com/bottlerocket-control:v0.4.0"
```

Still I can see node by kubectl but not by ssm?

**Comment by rverma-jm** [comment]
Also received an exception as 

```
  Normal   Scheduled               6m11s              default-scheduler                                       Successfully assigned spinnaker/atlantis-0 to ip-192-168-20-101.ap-south-1.compute.internal
  Warning  FailedCreatePodSandBox  6m6s               kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "3c9c615a5ad12ef6051373b03877e54b31e230ca1c98c810e2a8db22ff61c917": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  5m52s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "baadbc0fadab53f87f96429088c4469f72ff7ba9d07e66fbafc90eac93b248e4": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  5m37s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "db9aed4752e1514474725bf8abb20383eb6acb81ecb8d35479b109f76213f0a7": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  5m25s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "2c66ac89b773f6bbc148d9c2548600cb046d984dc1963f6c4157db0a5c7c0d00": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  5m14s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "fb05c7804543cf672b5ea96e25d4ecf50e4bc475c49a3606b77e939d289e602f": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  5m                 kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "cf24150d0b9d3345c76887f15ca94283692f12637f9db8c386d7a82ad73a3b2a": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  4m45s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "372c714a0dc79018845dbeef8556475a1c999172f22618c92976dce262614525": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  4m30s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "a425fd1f32bd4e408dd9d278694d237752d32ec72818e82913b54a646c2d2cb7": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  4m15s              kubelet, ip-192-168-20-101.ap-south-1.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "d7979acd99b477a5e9d535bc15ab46da5729fa59bfc7c28a3c3da7bd6c78afea": add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  19s (x17 over 4m)  kubelet, ip-192-168-20-101.ap-south-1.compute.internal  (combined from similar events): Failed create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "b5983eb39700a5df5cf8c0b34006364c67998f28a7ca366a1e6af31ca38c8ff4": add cmd: failed to assign an IP address to container
```,
I validated and updated coredns, aws-node and kube-proxy all running latest versions by eksctl.

**Comment by bcressey** [comment]
Metadata can't be overridden through user-data; it's filled with implementation details that can change over time.

Try setting the source directly:
```
[settings.host-containers.control]
source = "328549459982.dkr.ecr.us-east-1.amazonaws.com/bottlerocket-control:v0.4.0"
```

For the pod sandbox problem, it seems like `settings.kubernetes.pod-infra-container-image` might be incorrect for some reason. This may be a side effect of the invalid user-data, or a separate problem.

**Comment by jhaynes** [comment]
@rverma-jm We haven't heard from you in a bit, so I'm hoping this is resolved. If not, feel free to comment here or open a new issue if that makes more sense.
